### PR TITLE
chore: remove cli

### PR DESCRIPTION
**Pull request by guilhem-barthes**
## Related PR

- https://github.com/Substra/substra-backend/pull/667

## Summary

Remove `substra.cli` and `substra.sdk.Client.from_config_file()`.

## Notes

Fixes FL-489

## Please check if the PR fulfills these requirements

- [x] If necessary, the [changelog](https://github.com/Substra/substra/blob/main/CHANGELOG.md) has been updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] The commit message follows the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification
- For any breaking changes, companion PRs have been opened on the following repositories:
  - [ ] [substra-tests](https://github.com/Substra/substra)
  - [ ] [substrafl](https://github.com/Substra/substrafl)
  - [ ] [substra-documentation](https://github.com/Substra/substra-documentation)

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Review by ThibaultFy — COMMENTED** [review]
Thanks a lot, love the -1000 lines !!

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Removed
 
 - Function name from the Performance model ([#358](https://github.com/Substra/substra/pull/358))
+- `substra.cli` module ([#362](https://github.com/Substra/substra/pull/362))
+- `Client.from_config_file()`. The recommended way is to use the new config file format or env vars ([#362](https://github.com/Substra/substra/pull/362))
```
Comment (ThibaultFy) [inline CHANGELOG.md:25]:
```suggestion
- BREAKING: `substra.cli` module ([#362](https://github.com/Substra/substra/pull/362))
- BREAKING: `Client.from_config_file()`. The recommended way is to use the new config file format or env vars ([#362](https://github.com/Substra/substra/pull/362))
```

**Review by ThibaultFy — COMMENTED** [review]
Last comment and will be good :)

**Pull request merged**